ident/tag_iterator_matcher.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ident
+
+import (
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+)
+
+// TagIteratorMatcher is a gomock.Matcher that matches TagIterator
+type TagIteratorMatcher interface {
+	gomock.Matcher
+}
+
+// NewTagIteratorMatcher returns a new TagIteratorMatcher
+func NewTagIteratorMatcher(inputs ...string) TagIteratorMatcher {
+	if len(inputs)%2 != 0 {
+		panic("inputs must be specified in name-value pairs (i.e. divisible by 2)")
```

Comment:
nit: Perhaps just return `(TagIteratorMatcher, error)` instead of panic (I know this is used in tests, but even then a `require.NoError(t, err)` at least doesn't stop executing the rest of the package tests and fails cleanly.

Reply:
Btw a common pattern to deal w/ this is to have both `NewTagIteratorMatcher` that returns an error and a `MustNewTagIteratorMatcher` that panics on invalid input.

Reply:
Or maybe rename to `MustNew...` and panic

Reply:
Sure i'll do the two method approach.

Reply:
Done